File: ch4 supplemental/repl4.js
var hasChildren = prompt("Do you have children (Y/N)");

if(hasChildren != "N") 
{
	var childCount = prompt("How many children?");
	
	if(parseInt(childCount)!=NaN && isFinite(childCount)){
	console.log("You have " + childCount + " children");
	}
	
	else{
	console.log("You entered '" + childCount + "' which is an invalid response that can not be counted in the survey")
	}
	
}

console.log("Thanks for your participation, questionnaire complete.")

File: ch4 supplemental/repl4.cs
using System;
class MainClass {
  public static void Main (string[] args) {
    Console.WriteLine("Do you have children? (Y/N)");
    string hasChildren = Console.ReadLine();
    
    if(hasChildren != "N" || hasChilren != "n"){
    	int childCountAsInt =0;
    
    	Console.WriteLine("How many children?");
    	string childCount = Console.ReadLine();
    	
    	if(int.TryParse(childCount, out childCountAsInt)){
    		Console.WriteLine("You have " + childCountAsInt + " children.");
    	}
    	else{
    		Console.WriteLine("You entered an incorrect answer of '" + childCount + "', sorry.");
    	}
    }
    
    Console.WriteLine("Thanks for your participation, questionaire complete.");
    }
  } 
      